pennylane/optimize/gradient_descent.py

```diff
@@ -47,86 +48,110 @@ def update_stepsize(self, stepsize):
         """
         self._stepsize = stepsize
 
-    def step_and_cost(self, objective_fn, x, grad_fn=None):
+    def step_and_cost(self, objective_fn, *args, grad_fn=None, **kwargs):
         """Update x with one step of the optimizer and return the corresponding objective
         function value prior to the step.
 
         Args:
             objective_fn (function): the objective function for optimization
-            x (array): NumPy array containing the current values of the variables to be updated
+            *args : Variable length argument list for objective function
             grad_fn (function): Optional gradient function of the
                 objective function with respect to the variables ``x``.
                 If ``None``, the gradient function is computed automatically.
+                Must match shape of autograd derivative.
+            **kwargs : Variable length of keywords for the cost function
 
         Returns:
             tuple: the new variable values :math:`x^{(t+1)}` and the objective function output
                 prior to the step
         """
 
-        g, forward = self.compute_grad(objective_fn, x, grad_fn=grad_fn)
-        x_out = self.apply_grad(g, x)
+        g, forward = self.compute_grad(objective_fn, args, kwargs, grad_fn=grad_fn)
+        new_args = self.apply_grad(g, args)
 
         if forward is None:
-            forward = objective_fn(x)
+            forward = objective_fn(*args, **kwargs)
 
-        return x_out, forward
+        if len(new_args) == 1:
+            return new_args[0], forward
+        return new_args, forward
 
-    def step(self, objective_fn, x, grad_fn=None):
+    def step(self, objective_fn, *args, grad_fn=None, **kwargs):
         """Update x with one step of the optimizer.
 
         Args:
             objective_fn (function): the objective function for optimization
-            x (array): NumPy array containing the current values of the variables to be updated
+            *args : Variable length argument list for objective function
             grad_fn (function): Optional gradient function of the
                 objective function with respect to the variables ``x``.
                 If ``None``, the gradient function is computed automatically.
+                Must match shape of autograd derivative.
+            **kwargs : Variable length of keywords for the cost function
```

Comment:
```suggestion
            **kwargs : Variable length of keywords for the objective function
```
(for consistency with `*args`)